bpfman-operator/apis/v1alpha1/cluster_fentry_program_types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2024 The bpfman Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// ClFentryProgramInfo defines the Fentry program details
+type ClFentryProgramInfo struct {
+	ClFentryLoadInfo `json:",inline"`
+	// Whether the program should be attached to the function.
+	// +optional
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:default:={}
+	Links []ClFentryAttachInfo `json:"links"`
+}
+
+// ClFentryLoadInfo contains the program-specific load information for Fentry
+// programs
+type ClFentryLoadInfo struct {
+	// function is the name of the function to attach the Fentry program to.
+	Function string `json:"function"`
+}
+
+type AttachType string
+
+const (
+	Attach  AttachType = "true"
+	Dettach AttachType = "false"
+)
+
+// ClFentryAttachInfo indicates that the Fentry program should be attached to
+// the function identified in ClFentryLoadInfo. The only valid value for Attach
+// is true.
+type ClFentryAttachInfo struct {
+	// +kubebuilder:validation:Enum==Attach;Dettach;
+	Attach AttachType `json:"attach"`
```

Comment:
So from a user perspective the YAML for this field would look something like:
```yaml
attach: Attach | Detach
```

Is there a better name that can be used for this field? This is already in an `*AttachInfo` type so maybe something like `AttachmentMode` or just `Mode` is reasonable here?

Are there any configuration options that may be added in the future depending on the attachment mode?